hs-abci-sdk/src/Tendermint/SDK/Modules/Validators/Types.hs

```diff
@@ -0,0 +1,59 @@
+module Tendermint.SDK.Modules.Validators.Types where
+
+import           Control.Lens                           (Wrapped (_Wrapped'),
+                                                         iso, (^.), _Unwrapped')
+import qualified Data.Aeson                             as A
+import           Data.Bifunctor                         (Bifunctor (bimap, second))
+import           Data.ByteString                        (ByteString)
+import           Data.ByteString.Lazy                   (toStrict)
+import           Data.Either                            (fromRight)
+import           Data.ProtoLens                         (decodeMessage,
+                                                         encodeMessage)
+import           Data.Set                               (Set)
+import           Data.Text                              (pack)
+import           GHC.Generics                           (Generic)
+import           Network.ABCI.Types.Messages.FieldTypes (PubKey (PubKey),
+                                                         ValidatorUpdate)
+import           Tendermint.SDK.BaseApp                 (RawKey (..))
+import           Tendermint.SDK.Codec                   (HasCodec (..))
+
+
+data ValidatorsNameSpace
+
+type ValidatorsName = "validators"
+
+
+updatesListKey :: ByteString
+updatesListKey = "updatesList"
+
+validatorsMapKey :: ByteString
+validatorsMapKey = "validatorsMap"
+
+validatorsKeySetKey :: ByteString
+validatorsKeySetKey = "validatorsKeySet"
+
+
+newtype ValidatorUpdate_ = ValidatorUpdate_ ValidatorUpdate deriving (Eq, Generic)
+
+instance HasCodec ValidatorUpdate_ where
+  encode (ValidatorUpdate_ vu) = encodeMessage $ (vu ^. _Wrapped')
+  decode bs = bimap pack (ValidatorUpdate_ . (^. _Unwrapped')) $ decodeMessage bs
+
+newtype PubKey_ = PubKey_ PubKey deriving (Eq, Ord, Generic)
+
+instance RawKey PubKey_ where
+  rawKey = iso t f
+    where
+      t (PubKey_ p) = encodeMessage $ (p ^. _Wrapped')
+      f = PubKey_ . fromRight (PubKey "" "") . second (^. _Unwrapped') . decodeMessage
```

Comment:
This is fine, but as a note I'm sort of remembering that there is actually no instance where the `from` method of the iso is being used, we're almost always using it as a value of type `:: Lens' a ByteString`. This is why the `PubKey "" ""` seems awkward, I will get around to fixing this soon.

Reply:
See #255

Reply:
re `PubKey "" ""`: 
I figured this would be okay, since the only time it would be deserializing would be if it was reading the bytes from the db. And they must have been serialized from the pubkey to get put in the DB, so it _should_ never fail. Definitely is awkward thought...